File: app/src/main/java/com/kotlinlogin/java_mvp/LoginPresenter.java
package com.kotlinlogin.java_mvp;

import android.text.TextUtils;

import com.kotlinlogin.R;

public class LoginPresenter {

    private LoginContract loginView;

    LoginPresenter(LoginContract loginView) {
        this.loginView = loginView;
    }

    public void onLoginClick() {

        String userName = loginView.getUserName();
        String password = loginView.getPassword();

        if (TextUtils.isEmpty(userName)) {
            loginView.setUserNameError(R.string.err_username);
        } else if (TextUtils.isEmpty(password)) {
            loginView.setPasswordError(R.string.err_password);
        } else if (userName.equals(password)) {
            loginView.showLoginSuccess();
        } else {
            loginView.showLoginFailure();
        }
    }


}

File: app/src/main/java/com/kotlinlogin/kotlin_mvp/LoginPresenter.kt
package com.kotlinlogin.kotlin_mvp

import android.text.TextUtils
import com.kotlinlogin.R

class LoginPresenter(val contractView: LoginContractView) {

    fun clickLogin() {
        val userName: String = contractView.getUserName()
        val password: String = contractView.getPassword()
        if (TextUtils.isEmpty(userName)) {
            contractView.setUserError(R.string.err_username)
        } else if (TextUtils.isEmpty(password)) {
            contractView.setPasswordError(R.string.err_password)
        } else if (!userName.equals(password)) {
            contractView.showLoginFailure()
        } else {
            contractView.showLoginSuccess()
        }
    }
}